### ISSUE TITLE
[ANTLR] wrong definition of stretch type

### ISSUE THREAD

**Issue by hhorii** [Qiskit/openqasm#145]
Current stretch type is defined [as follows](https://github.com/Qiskit/openqasm/blob/0b3fe89214dedd6831926ee56ca5cbf599f423f5/source/grammar/qasm3.g4#L406-L409):
```
timingType
    : 'length'
    | 'stretch' Integer?
    ;
```
I think this is wrong because no space exists between `stretch` and `N` in [the spec](https://qiskit.github.io/openqasm/language/types.html#stretch).
> We distinguish different “orders” of stretch via stretchN types, where N is an integer between 0 to 255. stretch0 is an alias for the regular stretch.

The most naive (and not-smart) way to solve this issue is listing all the stretch types.
```
timingType
    : 'length'
    | 'stretch'
    | 'stretch0'
    | 'stretch1'
    ...
```
I'm not sure we can define`stretchN` with fragment `Digit*`.

**Comment by zachschoenfeld33** [comment]
There is no space. To create a space in ANTLR, you would need ' '. So `stretchN` would work. `N` is, admittedly, not limited between 0-255 though. An added example to fix this would be beneficial. Feel free to add a PR and tag me

**Comment by zachschoenfeld33** [comment]
Closed via #158
